scandium-core/src/main/java/org/eclipse/californium/scandium/DTLSConnector.java

```diff
@@ -225,6 +237,27 @@ public DTLSConnector(final DtlsConnectorConfig configuration, final SessionCache
 	}
 
 	/**
+	 * Sets the executor to use for processing records.
+	 * <p>
+	 * If this property is not set before invoking the {@linkplain #start() start method},
+	 * a new {@link StripedExecutorService} is used with a thread pool of size #(CPU cores).
+	 * <p>
+	 * In a production environment the thread pool used by the executor should be much larger
+	 * than just the number of CPU cores. This helps with performing multiple handshakes
+	 * in parallel, in particular if the key exchange requires look up of identities, e.g.
+	 * in a database or using a web service.
+	 * <p>
+	 * If this method is used to set an executor, the executor will <em>not</em> be shut down
+	 * by the {@linkplain #stop() stop method}.
+	 * 
+	 * @param executor The executor.
+	 */
+	public void setExecutor(StripedExecutorService executor) {
```

Comment:
Sorry for confusion. I disagree with my own comment. As we guarantee that executor field gets set before receiver worker thread gets started, we do not have any visibility issues with that. Start of a thread refreshes thread local memory.

I am not sure about some other fields, but I think we should postpone this discussion after refactoring of record processing.